src/zmalloc.c

```diff
@@ -436,9 +460,18 @@ char *zstrdup(const char *s) {
 }
 
 size_t zmalloc_used_memory(void) {
-    size_t um;
-    atomicGet(used_memory,um);
-    return um;
+    size_t local_num_active_threads;
+    size_t totol_mem = 0;
+    atomicGet(num_active_threads,local_num_active_threads);
+    if (local_num_active_threads > MAX_THREADS) {
+        local_num_active_threads = MAX_THREADS;
+    }
+    for (size_t i = 0; i < local_num_active_threads; ++i) {
+        long long thread_used_mem;
+        atomicGet(used_memory[i].used_memory, thread_used_mem);
+        totol_mem += thread_used_mem;
+    }
+    return totol_mem;
```

Comment:
@MeirShpilraien what about adding a assertion `assert(total_mem > 0)` to run a fully CI, and then finally delete it? or keep it

Reply:
I changed it to `long long` and added the assert, but I think we should remove it once we see the tests passes.

Reply:
fully CI (in progress): https://github.com/sundb/redis/actions/runs/10056811321

Reply:
Seems like it pass successfully